i6_models/assemblies/conformer/conformer.py

```diff
@@ -0,0 +1,110 @@
+from __future__ import annotations
+
+__all__ = ["ConformerBlockV1Config", "ConformerEncoderV1Config", "ConformerBlockV1", "ConformerEncoderV1"]
+
+import torch
+from torch import nn
+from dataclasses import dataclass
+
+from i6_models.config import ModelConfiguration
+from i6_models.parts.conformer import (
+    ConformerConvolutionV1,
+    ConformerConvolutionV1Config,
+    ConformerFrontendV1,
+    ConformerFrontendV1Config,
+    ConformerMHSAV1,
+    ConformerMHSAV1Config,
+    ConformerPositionwiseFeedForwardV1,
+    ConformerPositionwiseFeedForwardV1Config,
+)
+
+
+@dataclass
+class ConformerBlockV1Config(ModelConfiguration):
+    """
+    :param ff_cfg: Configuration for ConformerPositionwiseFeedForwardV1
+    :param mhsa_cfg: Configuration for ConformerMHSAV1
+    :param conv_cfg: Configuration for ConformerConvolutionV1
+    """
+
+    # nested configurations
+    ff_cfg: ConformerPositionwiseFeedForwardV1Config
+    mhsa_cfg: ConformerMHSAV1Config
+    conv_cfg: ConformerConvolutionV1Config
+
+
+@dataclass
+class ConformerEncoderV1Config(ModelConfiguration):
+    """
+    :param num_layers: Number of conformer layers in the conformer encoder
+    :param front_cfg: Configuration for ConformerFrontendV1
+    :param block_cfg: Configuration for ConformerBlockV1
+    """
+
+    num_layers: int
+
+    # nested configurations
+    front_cfg: ConformerFrontendV1Config
+    block_cfg: ConformerBlockV1Config
+
+
+class ConformerBlockV1(nn.Module):
+    """
+    Conformer block module
+    """
+
+    def __init__(self, cfg: ConformerBlockV1Config):
+        """
+        :param cfg: conformer block configuration with subunits for the different conformer parts
+        """
+        super().__init__()
+        self.ff_1 = ConformerPositionwiseFeedForwardV1(cfg=cfg.ff_cfg)
+        self.mhsa = ConformerMHSAV1(cfg=cfg.mhsa_cfg)
+        self.conv = ConformerConvolutionV1(model_cfg=cfg.conv_cfg)
+        self.ff_2 = ConformerPositionwiseFeedForwardV1(cfg=cfg.ff_cfg)
+        self.final_layer_norm = torch.nn.LayerNorm(cfg.ff_cfg.input_dim)
+
+    def forward(self, tensor: torch.Tensor):
+        """
+        :param tensor: input tensor of shape [B, T, F]
+        :return: torch.Tensor of shape [B, T, F]
+        """
+        residual = tensor  #  [B, T, F]
+        x = self.ff_1(residual)  #  [B, T, F]
+        residual = 0.5 * x + residual  #  [B, T, F]
+        x = self.mhsa(residual)  #  [B, T, F]
+        residual = x + residual  # [B, T, F]
+        x = self.conv(residual)  #  [B, T, F]
+        residual = x + residual  # [B, T, F]
+        x = self.ff_2(residual)  #  [B, T, F]
+        x = 0.5 * x + residual  #  [B, T, F]
+        x = self.final_layer_norm(x)  #  [B, T, F]
+        return x
+
+
+class ConformerEncoderV1(nn.Module):
```

Comment:
Other modules tend to have a docstring here. I would also add a link to the conformer paper somewhere. Maybe here would be a good place.